### PR TITLE
update files for winter 2023

### PR DESCRIPTION
**Pull request by eb-64-64**

### CODE REVIEW

**Comment by eb-64-64** [comment]
For reference, the tests are:
* Test 1: basic `get` and `insert` tests
* Test 2: basic `contains` tests
* Test 3: basic `operator[]` tests
* Test 4: basic `remove` tests
* Test 5: basic `clear` tests
* Test 6: basic `size` tests
* Test 7: rehashing test
* Test 8: word counting (to stress-test `operator[]`)
* Test 9: throw the dictionary at it (to stress-test `insert` and `remove`)

File: HashmapInterface.h [hunk 1]
```diff
@@ -1,89 +1,90 @@
...
-	/*
-	 * Return the value associated with the given key in the map
-	 *  Throw invalid_argument exception if key is not in map
-	 */
-	virtual int get(string key) const = 0;
+    /*
+     * Return the value associated with the given key in the map
+     *  Throw invalid_argument exception if key is not in map
+     */
+    virtual int get(const std::string &key) const = 0;
```
Comment (byubean) [inline HashmapInterface.h:32]: Don't we want `std::string const& key)`?

Would the current signature allow the function to reassign the reference to another const string?

Comment (byubean) [reply HashmapInterface.h:32]: I looked it up - I guess the options are the same. :)

File: Hashmap.h [hunk 2]
```diff
@@ -1,99 +0,0 @@
...
-#include <sstream>
-#include <iostream>
-#include <queue>
-#include "HashmapInterface.h"
-#define BUCKETS 10
-
-using namespace std;
-
-class Hashmap: public HashmapInterface
-{
```
Comment (byubean) [inline Hashmap.h:17]: Let's include this file with:

```c++
class Hashmap: public HashmapInterface {
  // and include stub overrides here
}
```

Comment (byubean) [reply Hashmap.h:17]: This way the project should at least compile "out of the box"

**Review by byubean — APPROVED** [review]

**Pull request merged**